### PR TITLE
Applied Simple Kotlin DSL Property Assignment

### PR DESCRIPTION
**Pull request by niyajali**
- https://blog.gradle.org/simpler-kotlin-dsl-property-assignment
- Enable Typesafe Project Accessors

### CODE REVIEW

File: kover-jvm-agent/build.gradle.kts [hunk 1]
```diff
@@ -23,17 +23,17 @@ plugins {
...
-    docsDirectory.set("jvm-agent")
-    description.set("Kover JVM instrumentation agent")
+    docsDirectory = "jvm-agent"
+    description = "Kover JVM instrumentation agent"
 }
 
 java {
-    sourceCompatibility = JavaVersion.VERSION_1_6
-    targetCompatibility = JavaVersion.VERSION_1_6
+    sourceCompatibility = JavaVersion.VERSION_1_8
```
Comment (shanshin) [inline kover-jvm-agent/build.gradle.kts:35]: We are specifically compiling for the `1.6` version for wider support of different JVMs.
Therefore, it is better to leave `1.6`

File: kover-offline-runtime/build.gradle.kts [hunk 2]
```diff
@@ -23,12 +23,12 @@ plugins {
...
 
 koverPublication {
-    description.set("Compiled dependency to ensure the operation of the code that has been instrumented offline")
+    description = "Compiled dependency to ensure the operation of the code that has been instrumented offline"
 }
 
 java {
-    sourceCompatibility = JavaVersion.VERSION_1_7
-    targetCompatibility = JavaVersion.VERSION_1_7
+    sourceCompatibility = JavaVersion.VERSION_1_8
```
Comment (shanshin) [inline kover-offline-runtime/build.gradle.kts:30]: It is also better to leave `1.7` here

File: kover-jvm-agent/build.gradle.kts (hunk 1 above)
Comment (niyajali) [reply kover-jvm-agent/build.gradle.kts:35]: But unable to run `clean build` locally, it says `error: Source option 6, 7 is no longer supported. Use 8 or later. `
I reverted back to as it is before, let see CI Build results.

**Review by shanshin — APPROVED** [review]
Thanks for the effort!

**Pull request merged**